File: regression/cs/tst6.cs
using Microsoft.Z3;
using System;

class Test
{

    public void Run()
    {        
        using (Context ctx = new Context())
        {
            Console.WriteLine(ctx.MkPower(ctx.MkReal(2), ctx.MkReal(1, 2)).Simplify());
            ctx.UpdateParamValue(":pp-decimal-precision", "50");
            ctx.UpdateParamValue(":pp-decimal", "true");
            Console.WriteLine(ctx.MkPower(ctx.MkReal(2), ctx.MkReal(1, 2)).Simplify());
            ctx.UpdateParamValue(":pp-decimal-precision", "20");
            Console.WriteLine(ctx.MkPower(ctx.MkReal(2), ctx.MkReal(1, 2)).Simplify());
        }
    }
}

File: old-regressions/python/tst6.py

# Copyright (c) 2015 Microsoft Corporation
from z3 import *

print(simplify(Sqrt(2)).sexpr())
set_option(":pp-decimal-precision", 50, pp_decimal=True)
print(simplify(Sqrt(2)).sexpr())
set_option(precision=20)
print(simplify(Sqrt(2)))
